src/routes/general_message.rs

```diff
@@ -5,25 +5,49 @@ use crate::siri_lite::{
     SiriResponse,
 };
 use crate::transit_realtime;
+use crate::utils;
 use actix::Addr;
 use actix_web::{error, AsyncResponder, Error, Json, Query, Result, State};
 use futures::future::Future;
 
-// #[derive(Deserialize, Debug)]
-// enum InfoChannel {
-//     Perturbation,
-//     Information,
-//     Commercial,
-// }
-
 #[derive(Deserialize, Debug)]
+#[serde(rename_all = "PascalCase")]
 pub struct Params {
-    // info_channel_ref: Vec<InfoChannel>,
+    /// start_time is the datetime from which we want the next departures
+    /// The default is the current time of the query
+    request_timestamp: Option<crate::siri_lite::DateTime>,
+}
+
+fn get_max_validity(
+    alert: &transit_realtime::Alert,
+    timezone: chrono_tz::Tz,
+) -> Option<crate::siri_lite::DateTime> {
+    alert
+        .active_period
+        .iter()
+        .filter_map(|p| utils::read_pbf_dt(p.end, timezone))
+        .max()
+        .map(|ndt| crate::siri_lite::DateTime(ndt))
 }
 
-fn display_alert(_alert: &transit_realtime::Alert) -> bool {
-    // TODO check active_period
-    true
+fn display_alert(
+    alert: &transit_realtime::Alert,
+    requested_dt: chrono::NaiveDateTime,
+    timezone: chrono_tz::Tz,
+) -> bool {
+    alert.active_period.iter().any(|p| {
+        if let Some(s) = utils::read_pbf_dt(p.start, timezone) {
+            if s > requested_dt {
```

Comment:
trop de if et de return, mais j’ai voulu proposer une meilleure réponse, et je trouve pas trop.

Reply:
```rust
match (utils::read_pbf_dt(p.start, timezone), utils::read_pbf_dt(p.end, timezone)) {
    (Some(s), _) if s > requested_dt => false,
    (_, Some(e)) if e < requested_dt => false,
    _ => true,
}
```

Reply:
```rust
utils::read_pbf_dt(p.start, timezone).map_or(true, |s| s <= requested_dt)
    && utils::read_pbf_dt(p.end, timezone).map_or(true, |e| requested_dt <= e)
```

Reply:
I changed it to the second version, but I'm not overly convinced it's more readable :roll_eyes: